src/lava/lib/optimization/solvers/generic/builder.py

```diff
@@ -0,0 +1,365 @@
+                                    problem: OptimizationProblem,
+                                    hyperparameters: ty.Dict[
+                                        str, ty.Union[int, npt.ArrayLike]]):
+        """Create __init__ method for the OptimizationSolverProcess class.
+
+        Parameters
+        ----------
+
+        problem: an instance of an OptimizationProblem defined by cost and
+        constraints which will be used to ensemble the necessary variables
+        and ports with their shape and initial values deriving from the
+        problem specification.
+        hyperparameters: A dictionary specifying values for steps_to_fire,
+        noise_amplitude, step_size and init_value. All but the last are
+        integers, the initial value is an array-like of initial values for the
+        variables defining the problem.
+
+        """
+
+        def constructor(self,
+                        hyperparameters: ty.Dict[
+                            str, ty.Union[int, npt.ArrayLike]],
+                        name: ty.Optional[str] = None,
+                        log_config: ty.Optional[LogConfig] = None) -> None:
+            super(type(self), self).__init__(hyperparameters=hyperparameters,
+                                             name=name,
+                                             log_config=log_config)
+            self.problem = problem
+            self.hyperparameters = hyperparameters
+            if not hasattr(problem, "variables"):
+                raise Exception(
+                    "An optimization problem must contain " "variables."
+                )
+            if hasattr(problem.variables, "continuous") or isinstance(
+                    problem.variables, ContinuousVariables
+            ):
+                self.continuous_variables = Var(
+                    shape=(problem.variables.continuous.num_vars, 2)
+                )
+            if hasattr(problem.variables, "discrete") or isinstance(
+                    problem.variables, DiscreteVariables
+            ):
+                self.discrete_variables = Var(
+                    shape=(
+                        problem.variables.num_variables,
+                        # problem.variables.domain_sizes[0]
+                    )
+                )
+            self.cost_diagonal = None
+            if hasattr(problem, "cost"):
+                self.cost_coefficients = SolverProcessBuilder._map_rank_to_coefficients_vars(
+                    problem.cost.coefficients
+                )
+                self.cost_diagonal = -problem.cost.coefficients[
+                    2].diagonal()
+            self.variable_assignment = Var(
+                shape=(problem.variables.num_variables,)
+            )
+            self.optimality = Var(shape=(1,))
+            self.feasibility = Var(shape=(1,))
+
+        self._process_constructor = constructor
+
+    def _create_model_constructor(self, target_cost: int):

```

Comment:
Such a method should be a general util of Lava, not hidden with lava-optimization. Just as general remark, no need to change it now.